src/System.IO.Compression/src/System/IO/Compression/DeflateStream.cs

```diff
@@ -531,6 +532,7 @@ protected override void Dispose(bool disposing)
                     }
                     finally
                     {
+                        ArrayPool<byte>.Shared.Return(_buffer);
```

Comment:
What happens if Dispose is called multiple times?  Will _buffer be returned multiple times?